test/testAdaptors/CurveAdaptor.t.sol

```diff
@@ -0,0 +1,1608 @@
+        );
+
+        registry.trustPosition(
+            fraxCrvUsdPoolPosition,
+            address(curveAdaptor),
+            abi.encode(FraxCrvUsdPool, FraxCrvUsdToken, FraxCrvUsdGauge, CurvePool.withdraw_admin_fees.selector)
+        );
+
+        registry.trustPosition(
+            mkUsdFraxUsdcPoolPosition,
+            address(curveAdaptor),
+            abi.encode(
+                mkUsdFraxUsdcPool,
+                mkUsdFraxUsdcToken,
+                mkUsdFraxUsdcGauge,
+                CurvePool.withdraw_admin_fees.selector
+            )
+        );
+
+        registry.trustPosition(
+            WethYethPoolPosition,
+            address(curveAdaptor),
+            abi.encode(WethYethPool, WethYethToken, WethYethGauge, CurvePool.withdraw_admin_fees.selector)
+        );
+
+        registry.trustPosition(
+            EthEthxPoolPosition,
+            address(curveAdaptor),
+            abi.encode(EthEthxPool, EthEthxToken, EthEthxGauge, CurvePool.withdraw_admin_fees.selector)
+        );
+
+        registry.trustPosition(
+            CrvUsdSdaiPoolPosition,
+            address(curveAdaptor),
+            abi.encode(CrvUsdSdaiPool, CrvUsdSdaiToken, CrvUsdSdaiGauge, CurvePool.withdraw_admin_fees.selector)
+        );
+
+        registry.trustPosition(
+            CrvUsdSfraxPoolPosition,
+            address(curveAdaptor),
+            abi.encode(CrvUsdSfraxPool, CrvUsdSfraxToken, CrvUsdSfraxGauge, CurvePool.withdraw_admin_fees.selector)
+        );
+
+        string memory cellarName = "Curve Cellar V0.0";
+        uint256 initialDeposit = 1e6;
+        uint64 platformCut = 0.75e18;
+
+        // Approve new cellar to spend assets.
+        address cellarAddress = deployer.getAddress(cellarName);
+        deal(address(USDC), address(this), initialDeposit);
+        USDC.approve(cellarAddress, initialDeposit);
+
+        bytes memory creationCode = type(Cellar).creationCode;
+        bytes memory constructorArgs = abi.encode(
+            address(this),
+            registry,
+            USDC,
+            cellarName,
+            cellarName,
+            usdcPosition,
+            abi.encode(0),
+            initialDeposit,
+            platformCut,
+            type(uint192).max
+        );
+        cellar = Cellar(deployer.deployContract(cellarName, creationCode, constructorArgs, 0));
+
+        cellar.addAdaptorToCatalogue(address(curveAdaptor));
+
+        USDC.safeApprove(address(cellar), type(uint256).max);
+
+        for (uint32 i = 2; i < 33; ++i) cellar.addPositionToCatalogue(i);
+        for (uint32 i = 2; i < 33; ++i) cellar.addPosition(0, i, abi.encode(true), false);
+
+        cellar.setRebalanceDeviation(0.030e18);
+
+        initialAssets = cellar.totalAssets();
+
+        slippageCoins.push(ERC20(address(0)));
+        slippageCoins.push(ERC20(address(0)));
+    }
+
+    // ========================================= HAPPY PATH TESTS =========================================
+
```

Comment:
A series of questions :)

1. Are we not caring about vCRV boosts working properly or not? We trust that it is and that the reward claiming that is carried out with this adaptor will capture any boosted rewards too if we implement abilities for the cellar to do vCRV voting and boosting?
 
2. What happens if zero is passed in as a param?
3. What about values larger than 1million?
4. Are there ways that pricing can be manipulated in a "stateful" way to attack through the curve adaptor? What happens if/when curve pool liquidity dries up? In general, we need mitigation methods.
5. Why are these varying amounts of tolerances used as params btw different happy path tests with different pools?